checks_windows.go

```diff
@@ -0,0 +1,410 @@
+}
+
+func (c cond) UserInGroup() (bool, error) {
+	users, err := wapi.LocalGroupGetMembers(c.Group)
+	if err != nil {
+		// Error is returned if group is empty.
+		return false, nil
+	}
+	for _, user := range users {
+		justName := strings.Split(user.Name, `\`)[1]
+		if c.User == user.Name || c.User == justName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (c cond) UserDetail() (bool, error) {
+	c.Key = strings.TrimSpace(c.Key)
+	lookingFor := false
+	if strings.ToLower(c.Key) == "yes" {
+		lookingFor = true
+	}
+	user, err := getLocalUser(c.User)
+	if err != nil {
+		return false, err
+	}
+	switch c.Key {
+	case "FullName":
+		if user.FullName == c.Key {
+			return true, nil
+		}
+	case "IsEnabled":
+		return user.IsEnabled == lookingFor, nil
+	case "IsLocked":
+		return user.IsLocked == lookingFor, nil
+	case "IsAdmin":
+		return user.IsAdmin == lookingFor, nil
+	case "PasswordNeverExpires":
+		return user.PasswordNeverExpires == lookingFor, nil
+	default:
+		fail("c.Key (" + c.Key + ") passed to userDetail is invalid.")
+		return false, errors.New("Invalid detail")
+	}
+	return false, nil
+}
+
+func (c cond) UserRights() (bool, error) {
+	// TODO consider /mergedpolicy when windows domain is active?
+	// domain support is untested, it should be easy to add a domain
+	// flag in the config though. then just make sure you're not getting
+	// invalid local policies instead of gpo
+
+	if c.User != "" && c.Group != "" {
+		fail("Can't have both User and Group for UserRights condition")
+		return false, errors.New("Bad condition config: conflicting keys User and Group")
+	} else if c.User != "" {
+		c.Group = c.User
+	} else if c.Group != "" {
+		c.User = c.Group
+	}
+
+	// TODO: only get section of users -- this can also falsely score correct for other secedit fields (like LegalNoticeText)
+	seceditOutput, err := getSecedit()
+	if err != nil {
+		return false, err
+	}
+
+	re := regexp.MustCompile("(?m)[\r\n]+^.*" + c.Value + ".*$")
+	privilegeString := strings.TrimSpace(string(re.Find([]byte(seceditOutput))))
+	debug("Privilege string for UserRights is:", privilegeString)
+	if privilegeString == "" {
+		return false, nil
+	}
+
+	if strings.Contains(privilegeString, c.User) {
+		// Sometimes, Windows just puts their user or group name instead of the
+		// SID. Really cool
+		return true, nil
+	}
+
+	privStringSplit := strings.Split(privilegeString, " ")
+	if len(privStringSplit) != 3 {
+		return false, errors.New("Error splitting privilege")
+	}
+
+	privStringSplit = strings.Split(privStringSplit[2], ",")
+	for _, sidValue := range privStringSplit {
+		sidValue = strings.TrimSpace(sidValue)
+		userForSid := strings.Split(sidToLocalUser(sidValue[1:]), "\\")
+		userSid := strings.TrimSpace(userForSid[0])
+		if len(userForSid) == 2 {
+			userSid = strings.TrimSpace(userForSid[1])
+		}
+		if userSid == c.User {
+			return true, nil
+		}
+	}
+
+	return false, err
+}
+
+func (c cond) WindowsFeature() (bool, error) {
+	return cond{
```

Comment:
Should this be modified if possible to be moved away from CommandOutput?

Reply:
Yes sir, it should. Similar to with `ShareExists` I think the Windows API migration fits best as a new issue/patch.